### DISCUSSION TITLE
increasing the number of latents

### DISCUSSION THREAD

**Discussion by apollo01gaming** [JCBrouwer/maua-stylegan2#10] [Q&A]
Hello, thank you for this repo,
I just finished rendering my video. and I want to increase the number of latents (generated pictures) dramatically. (Morphing )
How do I do that? 

I have looked into the other discussions made on this repo and I tried all of them. but can't quite make it work...

here is my test video. 
https://www.youtube.com/watch?v=gJFsWyW-2mQ

Thank you very much.

**Comment by JCBrouwer** [comment]
Hello, you'll want to take a look at adding looping then. There's two different strategies implemented in the library now which you can find [here](https://github.com/JCBrouwer/maua-stylegan2/blob/master/audioreactive/latent.py#L48).

If you create a large tensor of latents these functions can give you a sequence of latents that loops through them one-by-one.

You could, for example, replace the chroma latent sequence (which is used in the default examples) with this looping sequence and you'll have a video that morphs through your selected latents while still reacting to the beat.

[This example](https://github.com/JCBrouwer/maua-stylegan2/blob/master/audioreactive/examples/kelp.py) demonstrates the use of looping latents, although it is a little complicated as it generates these loops for the sections that are found automatically (by laplacian segmentation). You could try adapting it to your own music.

**Reply by apollo01gaming** [reply]
@JCBrouwer Hello, thank you for your reply. 
when i try to use kelp.py example. it giving me this error. 
FileNotFoundError: [Errno 2] No such file or directory: 'workspace/cyphept_kelp_drop_latents.npy'

**Reply by JCBrouwer** [reply]
Yeah, it's just what it says in the error: that file doesn't exist on your system. I chose some specific latents and saved them to the workspace/ folder. There's a couple other things like that hardcoded throughout the file, so you'll have to look for those and adjust them.

You can just generate some new latents there using [ar.generate_latents()](https://github.com/JCBrouwer/maua-stylegan2/blob/b9547a7f138e0a7bc212985a12b2dc325b77c32b/audioreactive/latent.py#L136) or just give it the latent_selection which was generated for you.
